noxfile.py

```diff
@@ -59,6 +108,30 @@ def run_pybamm_requires(session):
                 "advice.detachedHead=false",
                 external=True,
             )
+        if PYBAMM_ENV.get("PYBAMM_IDAKLU_EXPR_IREE") == "ON" and not os.path.exists(
+            "./iree"
+        ):
+            session.run(
+                "git",
+                "clone",
+                "--depth=1",
+                "--recurse-submodules",
+                "--shallow-submodules",
+                "--branch=candidate-20240507.886",
+                "https://github.com/openxla/iree",
+                "iree/",
+                external=True,
+            )
+            session.chdir("iree")
+            session.run(
+                "git",
+                "submodule",
+                "update",
+                "--init",
+                "--recursive",
+                external=True,
+            )
+            session.chdir("..")
```

Comment:
`session.chdir` can be used as a context manager, so something like
```suggestion
            with session.chdir("iree"):
                session.run(
                   "git",
                    "submodule",
                    "update",
                    "--init",
                    "--recursive",
                    external=True,
                )
```

should also work